src/classes/RD2_StatusMappingSettings_CTRL.cls

```diff
@@ -0,0 +1,534 @@
+        List<Mapping> mappings = new List<Mapping>();
+
+        try {
+            Map<String, RD2_StatusMapper.Mapping> mappingByStatus = statusMapper.getAll();
+
+            //get read-only mapping records first
+            for (String status : RD2_StatusMapper.READ_ONLY_STATUS_VALUES) {
+                RD2_StatusMapper.Mapping mapping = mappingByStatus.get(status);
+                if (mapping != null) {
+                    mappings.add(new Mapping(mapping));
+                }
+            }
+
+            //get other mapping records next
+            for (String status : mappingByStatus.keySet()) {
+                if (!RD2_StatusMapper.READ_ONLY_STATUS_VALUES.contains(status)) {
+                    mappings.add(new Mapping(mappingByStatus.get(status)));
+                }
+            }
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return mappings;
+    }
+
+
+    /**
+     * @description Saves status to state mapping records
+     * @param jsonMapping JSON representation of the admin defined mapping records
+     * @return String Deployment Id
+     */
+    @AuraEnabled
+    public static String saveMapping(String jsonMapping) {
+        return saveMapping(jsonMapping, new DeploymentHandler());
+    }
+
+    /**
+     * @description Saves status to state mapping records
+     * @param jsonMapping JSON representation of the admin defined mapping records
+     * @param deploymentHandler Deployment handler used to deploy CMT records and return the deployment Id
+     * @return String Deployment Id
+     */
+    @TestVisible
+    private static String saveMapping(String jsonMapping, MetadataDeploymentHandler deploymentHandler) {
+        try {
+            if (deploymentHandler.isInProgress()) {
+                throw new MappingException(System.Label.RD2_StatusMappingDeploymentAlreadyInProgress);
+            }
+
+            List<Metadata.CustomMetadata> records = buildCustomMetadataRecords(jsonMapping);
+            deploymentHandler.save(records);
+
+            //reset mapping record retrieval
+            statusMapper = null;
+
+        } catch (Exception e) {
+            throwAuraHandledException(e.getMessage());
+        }
+
+        return deploymentHandler.getDeploymentId();
+    }
+
+    /**
+     * @description Builds custom metadata records from status to state mapping records
+     * @param jsonMapping JSON representation of mapping records
+     * @return List<Metadata.CustomMetadata>
+     */
+    @TestVisible
+    private static List<Metadata.CustomMetadata> buildCustomMetadataRecords(String jsonMapping) {
+        List<Mapping> mappings = (List<Mapping>) JSON.deserialize(
+            jsonMapping, List<Mapping>.class
+        );
+
+        List<Metadata.CustomMetadata> records = new List<Metadata.CustomMetadata>();
+        for (Mapping mapping : mappings) {
+            records.add(buildCustomMetadata(mapping));
```

Comment:
Fourth attempt at this.

This all works perfectly as is, but I have a question.

Right now, this creates CMDT records for the 3 hard coded status values of Active, Lapsed and Closed. It's possible for a custom to manually edit those records in Salesforce setup. And, when they do those, changes show up in the new UI.
<img width="629" alt="image" src="https://user-images.githubusercontent.com/801017/82268307-b317f280-993c-11ea-98dc-66fe4fcce41a.png">

Luckily though, the validation and other services ignore the mappings for Active, Lapsed and Closed and use only the hard-coded internal mappings for those. My question is whether it's even worth creating CMDT records for these 3 hard-coded values at all. We definitely don't want customers to change what "Active" means this way. If they want to use a different label, they can easily override the label for "Active" without modifying the api name or meaning of it. Effectively, I think it might be better to just store the custom mappings without the 3 hard-coded ones.

Reply:
The change has been done. The Save action will store the Active, Lapsed and Closed Status mapping records.